src/features/decoration/decoration-placer.ts

```diff
@@ -0,0 +1,43 @@
+/********************************************************************************
+ * Copyright (c) 2020 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { injectable } from "inversify";
+import {
+    Decoration,
+    DecorationPlacer,
+    isSizeable,
+    ORIGIN_POINT,
+    Point,
+    SChildElement,
+    SModelElement,
+    SRoutableElement
+} from "sprotty";
+
+@injectable()
+export class GlspDecorationPlacer extends DecorationPlacer {
+
+    protected getPosition(element: SModelElement & Decoration): Point {
+        if (element instanceof SChildElement && element.parent instanceof SRoutableElement) {
+            return super.getPosition(element);
+        }
+        if (isSizeable(element))
+            return {
+                x: 12 * element.bounds.width,
```

Comment:
Where are these "magic numbers" coming from (12 rsp. 10). We should at least refactor them into static constants